pkg/queue/breaker_test.go

```diff
@@ -180,10 +227,20 @@ func TestSemaphore_AddCapacityLessThenReducers(t *testing.T) {
 	sem.Acquire()
 	sem.ReduceCapacity(2)
 	assertEqual(int32(2), sem.reducers, t)
-	sem.AddCapacity(3)
+	sem.Release()
+	sem.Release()
+	sem.Release()
 	assertEqual(int32(0), sem.reducers, t)
 }
 
+func TestSemaphore_AddCapacityOverflow(t *testing.T) {
+	sem := NewSemaphore(2, 2)
+	sem.Acquire()
+	sem.Acquire()
+	response := sem.AddCapacity(3)
+	assertEqual(ErrAddCapacity, response, t)
```

Comment:
Should this test include that Capacity is added until full and then it returns an error?